src/components/nabla/Pools/Swap/AddLiquidity/useAddLiquidity.ts

```diff
@@ -1,60 +1,102 @@
+import { useCallback, useMemo } from 'preact/hooks';
+import Big from 'big.js';
+import { useQueryClient } from '@tanstack/react-query';
+
 import { swapPoolAbi } from '../../../../../contracts/nabla/SwapPool';
-import { useGetAppDataByTenant } from '../../../../../hooks/useGetAppDataByTenant';
 import { useModalToggle } from '../../../../../services/modal';
-import { decimalToNative, FixedU128Decimals } from '../../../../../shared/parseNumbers/metric';
-import { useContractBalance } from '../../../../../shared/useContractBalance';
-import { useContractWrite } from '../../../../../shared/useContractWrite';
-import schema from './schema';
-import { AddLiquidityValues } from './types';
+import { decimalToRaw } from '../../../../../shared/parseNumbers/metric';
+import { erc20WrapperAbi } from '../../../../../contracts/nabla/ERC20Wrapper';
+import { useErc20ContractBalance } from '../../../../../hooks/nabla/useErc20ContractBalance';
+import { useContractWrite } from '../../../../../hooks/nabla/useContractWrite';
+import { cacheKeys } from '../../../../../constants/cache';
+
+export type AddLiquidityValues = {
+  amount: string;
+};
+
+const schema = Yup.object<AddLiquidityValues>().shape({
+  amount: Yup.number().positive().required(),
+});
 
-export const useAddLiquidity = (poolAddress: string, tokenAddress: string) => {
+export const useAddLiquidity = (
+  poolAddress: string,
+  tokenAddress: string,
+  poolTokenDecimals: number,
+  lpTokenDecimals: number,
+) => {
   const queryClient = useQueryClient();
-  const { indexerUrl } = useGetAppDataByTenant('nabla').data || {};
   const toggle = useModalToggle();
 
-  const balanceQuery = useContractBalance({ contractAddress: tokenAddress, decimals: FixedU128Decimals });
-  const depositQuery = useContractBalance({
+  const balanceQuery = useErc20ContractBalance(erc20WrapperAbi, {
+    contractAddress: tokenAddress,
+    decimals: poolTokenDecimals,
+  });
+
+  const depositQuery = useErc20ContractBalance(swapPoolAbi, {
     contractAddress: poolAddress,
-    abi: swapPoolAbi,
-    decimals: FixedU128Decimals,
+    decimals: lpTokenDecimals,
   });
 
   const form = useForm<AddLiquidityValues>({
     resolver: yupResolver(schema),
-    defaultValues: {},
+    defaultValues: {
+      amount: undefined,
+    },
   });
 
   const mutation = useContractWrite({
     abi: swapPoolAbi,
     address: poolAddress,
     method: 'deposit',
-    onError: () => {
-      // ? log error - alert not needed as the transaction modal displays the error
-    },
-    onSuccess: () => {
-      form.reset();
-      balanceQuery.refetch();
-      depositQuery.refetch();
-      queryClient.refetchQueries([cacheKeys.swapPools, indexerUrl]);
+    mutateOptions: {
+      onError: () => {
+        // ? log error - alert not needed as the transaction modal displays the error
+      },
+      onSuccess: () => {
+        form.reset();
+        balanceQuery.refetch();
+        depositQuery.refetch();
+        setTimeout(() => {
```

Comment:
Could we apply the same fix about not using setTimeout here as well? It would help us maintain consistency and avoid potential issues across the application.

Reply:
Same comment here, this is the indexer again.